dap/adapter.go

```diff
@@ -156,6 +156,37 @@ func (d *Adapter[C]) Next(c Context, req *dap.NextRequest, resp *dap.NextRespons
 	return nil
 }
 
+func (d *Adapter[C]) StepIn(c Context, req *dap.StepInRequest, resp *dap.StepInResponse) error {
+	var (
+		subReq  dap.NextRequest
+		subResp dap.NextResponse
+	)
+
+	subReq.Arguments.ThreadId = req.Arguments.ThreadId
+	subReq.Arguments.SingleThread = req.Arguments.SingleThread
+	subReq.Arguments.Granularity = req.Arguments.Granularity
+
+	if err := d.Next(c, &subReq, &subResp); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Adapter[C]) StepOut(c Context, req *dap.StepOutRequest, resp *dap.StepOutResponse) error {
+	var (
+		subReq  dap.ContinueRequest
+		subResp dap.ContinueResponse
+	)
+
+	subReq.Arguments.ThreadId = req.Arguments.ThreadId
+	subReq.Arguments.SingleThread = req.Arguments.SingleThread
+
+	if err := d.Continue(c, &subReq, &subResp); err != nil {
```

Comment:
Implementation looks good to me. 
But maybe leaving them as nop is enough?

Reply:
They show up in the visual studio code UI and don't seem to be behind a capability so I thought aliasing them was probably easiest for now. If also like to get done feedback from users about how they feel these buttons should work.

If it doesn't make sense we can leave it out but I figured that step in is the equivalent of next anyway if there's no function and step out is the equivalent of continue if you consider llb to be one big main function.

Reply:
Thanks for the explanation, making them as aliases sounds good to me.